onnx/test/shape_inference_test.py

```diff
@@ -69,6 +69,7 @@ def _compare_value_infos(self, vi_type: TypeProto, inferred_vi_type: TypeProto)
             assert vi_type.tensor_type.elem_type == inferred_vi_type.tensor_type.elem_type
             assert vi_type.tensor_type.HasField('shape') == inferred_vi_type.tensor_type.HasField('shape')
             if vi_type.tensor_type.HasField('shape'):
+                assert len(vi_type.tensor_type.shape.dim) == len(inferred_vi_type.tensor_type.shape.dim)
```

Comment:
Previously if the dim_size of expected value_info is zero, the dim_size of inferred value_info can be anything because the following loop will never run. Add this line here to enhance the dim comparison.